docker/table-diff-handle/tablediffview/create_diff_markdown.py

```diff
@@ -0,0 +1,149 @@
+import csv
+import html2text
+import re
+from tablediffview.config import CONFIG
+from itertools import zip_longest
+from os import listdir
+from os.path import isfile
+from tabulate import tabulate
+
+
+class htmlcelldata:
+    def __init__(self, text: str, align: str) -> None:
+        self.text = text
+        self.align = align
+
+
+class diffmarkdown:
+    def __init__(self, test_file: str, table: str, md: str) -> None:
+        self.test_file = test_file
+        self.table = table
+        self.md = md
+
+
+def create_diffs(diff_dir: str) -> list[diffmarkdown]:
+    test_file_diffs = get_test_file_diffs(diff_dir)
+    csv_path_pattern = "{0}/{1}/{2}_{3}.csv"
+    mds: list[diffmarkdown] = []
+    for test_file, diff_tables in test_file_diffs.items():
+        for table_name in diff_tables:
+            print("Create diff view for: " + table_name)
+            with open(
+                csv_path_pattern.format(diff_dir, test_file, table_name, "current"),
+                encoding="utf-8",
+            ) as current:
+                with open(
+                    csv_path_pattern.format(
+                        diff_dir, test_file, table_name, "reference"
+                    ),
+                    encoding="utf-8",
+                ) as reference:
+                    diff_md = tabulate(
+                        create_diff_table(current, reference),
+                        headers="firstrow",
+                        tablefmt="github",
+                    )
+                    mds.append(diffmarkdown(test_file, table_name, diff_md))
+
+    return mds
+
+
+# Get the list of table, which haved changed
+def get_test_file_diffs(diff_dir: str) -> dict[str, set[str]]:
+    result = {}
+    csv_current_pattern = re.compile(".+_current.csv")
+    for dir in listdir(diff_dir):
+        tables = set()
+        test_file_dir = f"{diff_dir}/{dir}"
+        if not isfile(test_file_dir):
+            for f in listdir(test_file_dir):
+                if isfile(f"{test_file_dir}/{f}") and csv_current_pattern.match(f):
+                    tables.add(str(f).replace("_current.csv", ""))
+            result[f"{str(dir)}"] = tables
+    return result
+
+
+def create_diff_table(c_list, r_list):
+    c_list = list(csv.reader(c_list, delimiter=";", skipinitialspace=True))
+    r_list = list(csv.reader(r_list, delimiter=";", skipinitialspace=True))
+    diff_table = []
+    create_diff_rows(diff_table, c_list, r_list)
+    return diff_table
+
+
+def create_diff_rows(
+    diff_table: list[list[str]], c_list: list[list[str]], r_list: list[list[str]]
```

Comment:
@TruongQuangSB Same here 

```suggestion
    diff_table: list[list[str]], changed_list: list[list[str]], reference_list: list[list[str]]
```

Reply:
ok